pkg/backup/backup/backup_cleaner.go

```diff
@@ -55,11 +57,23 @@ func (bc *backupCleaner) Clean(backup *v1alpha1.Backup) error {
 	}
 	ns := backup.GetNamespace()
 	name := backup.GetName()
+	backupJobName := backup.GetBackupJobName()
+
+	klog.Infof("start to ensure that backup %s/%s job %s have finished", ns, name, backupJobName)
+
+	finished, err := bc.ensureBackupJobFinished(backup)
+	if err != nil {
+		return fmt.Errorf("ensure %s/%s job %s finished failed: %s", ns, name, backupJobName, err)
+	}
+	if !finished {
+		klog.Infof("wait for backup %s/%s job %s to finish", ns, name, backupJobName)
```

Comment:
Should we return error to trigger resync here?

Reply:
Use informer to watch job in BackupController, so when the backup job finish, it will trigger resync.

Reply:
But you only handle the delete event now, when the job is finished, it will not trigger resync, the code implemented now can trigger resync when you delete the job in the `Clean` function, anyway, we also have resync period with 30s by default.

Reply:
Function `ensureBackupJobFinished` will delete the job if it is running. It means that if the return `finished` is false, the job will be deleted.

Reply:
ok